benchmark/script.py

```diff
@@ -205,6 +206,31 @@ def get_cluster(options, env):
     return cluster
 
 
+def add_userdata(options, result):
+    """
+    Parse the userdata option and add to result.
+
+    :param BenchmarkOptions options: Script options.
+    :param dict result: Result dictionary.
+    """
+    userdata = options['userdata']
+    if userdata:
+        try:
+            if userdata.startswith('@'):
+                try:
+                    with open(userdata[1:]) as f:
+                        result['userdata'] = json.load(f)
+                except IOError as e:
+                    usage(
+                        options,
+                        'Invalid user data file: {}'.format(e.strerror)
+                    )
+            else:
+                result['userdata'] = json.loads(userdata)
+        except ValueError as e:
+            usage(options, 'Invalid user data: {}'.format(e.args[0]))
```

Comment:
:question: I would strongly consider changing this function so that it _doesn't_ take `result` and instead just returns the userdata extracted from `BenchmarkOptions` (or even just take `options['userdata']` as a parameter). Not only will it simplify the function, but it will also simplify the tests.